lms/djangoapps/courseware/views/index.py

```diff
@@ -165,6 +166,9 @@ def _setup_masquerade_for_effective_user(self):
             self.is_staff,
             reset_masquerade_data=True
         )
+        _, is_enrolled = CourseEnrollment.enrollment_mode_for_user(self.effective_user, self.course_key)
+        if not is_enrolled:
+            self.effective_user = AnonymousUser()
```

Comment:
The check for enrollment happens inside `check_course_access()` and we do not want to call this directly (the bugfix PR that this is based on actually fixed this exact bug in `get()` above). Since we have already checked this above we can simply do something like:

```
if self.view == PUBLIC_VIEW:
  self.masquerade, self.effective_user = None, AnonymousUser()
else:
  self.masquerade, self.effective_user = setup_masquerade(
```

Reply:
Makes sense. I've updated it accordingly.